### PR TITLE
chore: refactor auth to async await

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
- Refactor auth handling to use `async/await` instead of promises
- Slightly refactor to simplify + for readability

#### Where should the reviewer start?
`auth/index.ts`

#### How should this be manually tested?
- Build with `npm run build` & then test out using `snyk auth`

### CODE REVIEW

File: src/cli/commands/auth/index.ts [hunk 1]
```diff
@@ -12,61 +12,56 @@ import { MisconfiguredAuthInCI } from '../../../lib/errors/misconfigured-auth-in
...
-    console.log(msg);
-  } else {
-    return Promise.reject(MisconfiguredAuthInCI());
-  }
-
-  const lbl = 'Waiting...';
+    urlStr + '\n');
 
-  return spinner(lbl).then(() => {
+  try {
```
Comment (kyegupov) [inline src/cli/commands/auth/index.ts]: you are dropping spinner here? Why?

File: src/cli/commands/auth/index.ts [hunk 2]
```diff
@@ -12,61 +12,56 @@ import { MisconfiguredAuthInCI } from '../../../lib/errors/misconfigured-auth-in
...
-
 const apiUrl = url.parse(config.API);
 const authUrl = apiUrl.protocol + '//' + apiUrl.host;
 const debug = Debug('snyk-auth');
 let attemptsLeft = 0;
 
+module.exports = auth;
+
 function resetAttempts() {
   attemptsLeft = 30;
```
Comment (kyegupov) [inline src/cli/commands/auth/index.ts]: 30 is a lot. Do you know how was this number chosen?

Comment (lili2311) [reply src/cli/commands/auth/index.ts]: I don't perhaps worth asking in slack, it does seem like a lot

File: src/cli/commands/auth/index.ts (hunk 1 above)
Comment (lili2311) [reply src/cli/commands/auth/index.ts]: Two reasons:
- it prints `Waiting...` and then is clears it a few seconds later. Seems not very helpful
- the way previous code was written means it cleared the spinner after getting the value back, i a promise chain couldn't see a nice way to add this into the refactored code as we return the value.

Comment (kyegupov) [reply src/cli/commands/auth/index.ts]: you could return "testAuthComplete().then(spinner.clear)"

But if you feel that the spinner was useless, LET IT DIE.

**Pull request merged**